pinot-controller/src/test/java/com/linkedin/pinot/controller/helix/sharding/SegmentAssignmentStrategyTest.java

```diff
@@ -171,10 +191,189 @@ public void testBalanceNumSegmentAssignmentStrategy() throws Exception {
     _helixAdmin.dropResource(HELIX_CLUSTER_NAME, TableNameBuilder.OFFLINE.tableNameWithType(TABLE_NAME_BALANCED));
   }
 
+  @Test
+  public void testTableLevelAndMirroringReplicaGroupSegmentAssignmentStrategy() throws Exception {
+    // Create the configuration for segment assignment strategy.
+    ReplicaGroupStrategyConfig replicaGroupStrategyConfig = new ReplicaGroupStrategyConfig();
+    replicaGroupStrategyConfig.setNumInstancesPerPartition(3);
+    replicaGroupStrategyConfig.setMirrorAssignmentAcrossReplicaGroups(true);
+
+    // Create table config
+    TableConfig tableConfig = new TableConfig.Builder(CommonConstants.Helix.TableType.OFFLINE)
+        .setTableName(TABLE_NAME_TABLE_LEVEL_REPLICA_GROUP)
+        .setNumReplicas(NUM_REPLICA)
+        .setSegmentAssignmentStrategy("ReplicaGroupSegmentAssignmentStrategy")
+        .build();
+
+    tableConfig.getValidationConfig().setReplicaGroupStrategyConfig(replicaGroupStrategyConfig);
+
+    // Create the table and upload segments
+    _pinotHelixResourceManager.addTable(tableConfig);
+    Thread.sleep(3000);
```

Comment:
Before you decide that a sleep is unnecessary, please validate why the sleep was there in the first place. Ideally, the code should be to busy loop waiting for that event. Hopefully the author who put in the sleep wrote a comment on what event we are waiting for there.